### PR TITLE
Scale LH Prover

### PR DESCRIPTION
**Pull request by preethamr**
## Type of change

- [ ] Docs change / dependency upgrade
- [ ] Configuration / tooling changes
- [ ] Refactoring
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires changes in customer code

## Related Issue(s)

Fixes

### CODE REVIEW

File: packages/agents/lighthouse/src/config.ts [hunk 1]
```diff
@@ -160,6 +162,8 @@ export const getEnvConfig = (
       configJson.relayerWaitTime ||
       configFile.relayerWaitTime ||
       DEFAULT_RELAYER_WAIT_TIME,
+    concurrency:
+      process.env.NXTP_PROVER_CONCURRENCY || configJson.concurrency || configFile.concurrency || DEFAULT_CONCURRENCY,
```
Comment (wanglonghong) [inline packages/agents/lighthouse/src/config.ts:166]: In general, the data type of env variable is string but concurrency is number so need a fix here

**Review by wanglonghong — COMMENTED** [review]
I am still skeptical about this solution how it could improve the messaging layer processing because as we know nodejs is a single threaded application. 
The one way I am thinking is to make lh prover pub/sub as we do in sequencer. but need others input here

**Comment by rhlsthrm** [comment]
> I am still skeptical about this solution how it could improve the messaging layer processing because as we know nodejs is natively a single threaded application. The one way I am thinking is to make lh prover pub/sub as we do in sequencer. but need others input here

This would still work because even though Node is single threaded it will work on all the processes at the same time. So when it's waiting on one, it will process the other.

File: packages/agents/lighthouse/src/config.ts (hunk 1 above)
Comment (rhlsthrm) [reply packages/agents/lighthouse/src/config.ts:166]: I think this is correct.

Comment (preethamr) [reply packages/agents/lighthouse/src/config.ts]: True. Pushed one way check and convert, perhaps there is a more elegant way to do this.

Interesting that we have this pattern at a few different places like  for `relayerWaitTime` , `NXTP_CARTOGRAPHER_POLL_INTERVAL`. 
We should update this in all places right @rhlsthrm @wanglonghong  ?

Comment (wanglonghong) [reply packages/agents/lighthouse/src/config.ts]: Yes right. we have similar things in other places.

Comment (rhlsthrm) [reply packages/agents/lighthouse/src/config.ts]: Correct I think it's working because we are using the defaults, so we need this if we override it.

**Review by wanglonghong — APPROVED** [review]

**Pull request merged**